src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/Forwarder.cs

```diff
@@ -37,12 +37,29 @@ private bool TryRehydrateMarshalAsAttribute(TypePositionInfo info, out Attribute
                 return true;
             }
 
-            if (info.MarshallingAttributeInfo is NativeContiguousCollectionMarshallingInfo collectionMarshalling
-                && collectionMarshalling.UseDefaultMarshalling
-                && collectionMarshalling.ElementCountInfo is NoCountInfo or SizeAndParamIndexInfo
-                && collectionMarshalling.ElementMarshallingInfo is NoMarshallingInfo or MarshalAsInfo { UnmanagedType: not UnmanagedType.CustomMarshaler }
-                && info.ManagedType is IArrayTypeSymbol)
+            if (info.ManagedType is SzArrayType)
             {
+                CountInfo countInfo;
+                MarshallingInfo elementMarshallingInfo;
+                if (info.MarshallingAttributeInfo is NativeContiguousCollectionMarshallingInfo collectionMarshalling
+                    && collectionMarshalling.UseDefaultMarshalling
+                    && collectionMarshalling.ElementCountInfo is NoCountInfo or SizeAndParamIndexInfo
+                    && collectionMarshalling.ElementMarshallingInfo is NoMarshallingInfo or MarshalAsInfo { UnmanagedType: not UnmanagedType.CustomMarshaler }
+                    )
+                {
+                    countInfo = collectionMarshalling.ElementCountInfo;
+                    elementMarshallingInfo = collectionMarshalling.ElementMarshallingInfo;
+                }
+                else if (info.MarshallingAttributeInfo is MissingSupportCollectionMarshallingInfo missingSupport)
+                {
+                    countInfo = missingSupport.CountInfo;
+                    elementMarshallingInfo = missingSupport.ElementMarshallingInfo;
+                }
+                else
+                {
+                    return false;
```

Comment:
Can we document this case. It isn't obvious how we got here.

Reply:
```suggestion
                    // This condition can be hit in two ways:
                    // 1. User uses the MarshalUsing attribute to provide count info or element marshalling information.
                    // Since the MarshalUsing attribute doesn't exist on downlevel platforms where we don't support arrays, this case is unlikely to come in supported scenarios, but could come up with a custom CoreLib implementation
                    // 2. User provides a MarsalAs attribute with the ArraySubType field set to UnmanagedType.CustomMarshaler 
                    // As mentioned above, we don't support ICustomMarshaler in the generator so we fail to forward the attribute instead of partially fowarding it.
                    return false;
```